test/kv679_eqc.erl

```diff
@@ -0,0 +1,493 @@
+%% -------------------------------------------------------------------
+%%
+%% eqc model/tests for kv679 type bugs
+%%
+%% Copyright (c) 2017 bet365, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(kv679_eqc).
+
+-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+-include_lib("eqc/include/eqc_statem.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-compile(export_all).
+
+-define(P1, p1).
+-define(REPLICAS, [p2, p3, f1, f2, f3]).
+
+-record(state, {
+          pfhead=?P1,
+          %% Actor Ids for replicas in the system
+          replicas= ?REPLICAS,
+          %% grow list of values put with their contexts, to decide
+          %% what values should be left at the end
+          values=[],
+          %% result of last get
+          last_get=undefined
+         }).
+
+-record(replica, {
+          id, %% The replica ID
+          epoch=0,
+          data=orddict:new() %% just use an orddict for now
+         }).
+
+-define(BUCKET, <<"b">>).
+%% The set of keys, we want more than one, but not too many (but start
+%% with one while developing)
+-define(KEYS, [<<"A">>]).
+
+-define(NUMTESTS, 100).
+-define(QC_OUT(P),
+        eqc:on_output(fun(Str, Args) ->
+                              io:format(user, Str, Args) end, P)).
+
+eqc_test_() ->
+    {setup,
+     fun() ->
+             meck:new(riak_core_bucket),
```

Comment:
Will do, I added `setup` and `teardown` after the `eqc_test` function, didn't refactor.